pkg/cortex/modules.go

```diff
@@ -169,8 +169,6 @@ func (t *Cortex) initDistributor() (serv services.Service, err error) {
 }
 
 func (t *Cortex) initQuerier() (serv services.Service, err error) {
-	queryable, engine := querier.New(t.Cfg.Querier, t.Overrides, t.Distributor, t.StoreQueryables, t.TombstonesLoader, prometheus.DefaultRegisterer)
```

Comment:
Design wise, looks a bit weird to me that the querier is not created by `initQuerier()`. I'm probably missing something, but wouldn't possible to rollback this change and just have the Ruler module depending on the Querier so that it can reference `t.Queryable` and `t.Engine`?